pkgs/tools/virtualization/nixos-container/nixos-container.pl

```diff
@@ -201,10 +203,17 @@ sub writeNixOSConfig {
         my $nixosConfigFile = "$root/etc/nixos/configuration.nix";
         writeNixOSConfig $nixosConfigFile;
 
-        system("nix-env", "-p", "$profileDir/system",
-               "-I", "nixos-config=$nixosConfigFile", "-f", "$nixenvF",
-               "--set", "-A", "system") == 0
-            or die "$0: failed to build initial container configuration\n";
+        if (defined $nixpkgsPath) {
+            system("nix-env", "-p", "$profileDir/system",
+                   "-I", "nixos-config=$nixosConfigFile", "-I", "nixpkgs=$nixpkgsPath",
```

Comment:
It seems unfortunate to duplicate this block only to add one option. Can you try deduplicating it?

Reply:
I considered this, but I'm not very familiar with Perl. Does it allow putting a function's arguments into an array and then passing that? Similar to Python's f(*args).

Reply:
Ok, got this done as an args array.